aws/hll/aws-s3-transfer-manager/Cargo.toml

```diff
@@ -0,0 +1,31 @@
+[package]
+name = "aws-s3-transfer-manager"
+version = "0.1.0"
+edition = "2021"
+authors = ["AWS Rust SDK Team <aws-sdk-rust@amazon.com>", "Aaron Todd <todaaron@amazon.com>"]
+description = "S3 Transfer Manager"
+license = "Apache-2.0"
+repository = "https://github.com/smithy-lang/smithy-rs"
+publish = false
+
+
+[dependencies]
+aws-sdk-s3 = { version = "1.22.0", features = ["behavior-version-latest", "test-util"] }
+aws-smithy-http = "0.60.7"
+tracing = "0.1"
+async-trait = "0.1.80"
+# FIXME - upgrade to hyper 1.x
+hyper = { version = "0.14.28", features = ["client"] }
+aws-smithy-types = "1.1.8"
+bytes = "1"
+thiserror = "1.0.58"
```

Comment:
Do we use `thiserror`? IIRC, it incurs additional compilation time that some customers may not be happy with ([related thread](https://www.reddit.com/r/rust/comments/gj8inf/rust_structuring_and_handling_errors_in_2020/)). No need for us to make a decision as part of this PR, since it'll be merged into a feature branch, but we probably want to compare compilation time at some point in the future with/without using `thiserror`.

Reply:
Yeah I think we will revisit the overall structure of errors and what we return from this library. I pulled in `thiserror` to make it easier to get started. I imagine we'll rip it out when we revisit errors more holistically.

Reply:
Hmm, although it's already in the dependency tree from `aws-smithy-protocol-test`, so I suppose using `thiserror` here isn't a big deal.
```
thiserror v1.0.61
├── aws-s3-transfer-manager v0.1.0 (/Users/awsaito/src/smithy-rs/aws/hll/aws-s3-transfer-manager)
└── aws-smithy-protocol-test v0.60.7
```

Reply:
Although, since it's already in the dependency tree from `aws-smithy-protocol-test`, I suppose using `thiserror` here isn't a big deal.
```
thiserror v1.0.61
├── aws-s3-transfer-manager v0.1.0 (src/smithy-rs/aws/hll/aws-s3-transfer-manager)
└── aws-smithy-protocol-test v0.60.7
    └── aws-smithy-runtime v1.5.4
        ├── aws-config v1.5.0
...
```

Reply:
Probably profile it before doing anything but we'll be revisiting errors anyway once we get a feel for what errors should be exposed and which should be opaque. I wasn't sure yet so defaulted to exposing errors rather than wrapping them into something opaque.